quetz/frontend.py

```diff
@@ -102,13 +110,33 @@ def render_index(config):
 
 
 @catchall_router.get('/{resource:path}', include_in_schema=False)
-def static(resource: str):
+def static(
+    resource: str,
+    session: dict = Depends(get_session),
+    dao: Dao = Depends(get_dao),
+    auth: authorization.Rules = Depends(get_rules),
+):
+    user_id = auth.assert_user()
+    global profile
+    profile = dao.get_profile(user_id)
+    profile.user.id = str(uuid.UUID(bytes=profile.user.id))
+
+    get_rendered_index(config_data, profile, index_template)
+
     if "." not in resource:
         return FileResponse(path=os.path.join(frontend_dir, "index.html"))
     else:
         return FileResponse(path=os.path.join(frontend_dir, resource))
 
 
+def get_rendered_index(config_data, profile, index_template):
+    config_data["logged_in_user_profile"] = rest_models.Profile.from_orm(profile).json()
+    logger.info(f"Page config: {config_data}")
+    index_rendered = index_template.render(page_config=config_data)
+    with open(static_dir / "index.html", "w") as fo:
```

Comment:
this is writing out the index.html, but it changes per user. we need to return it with a different response type (HTMLResponse).